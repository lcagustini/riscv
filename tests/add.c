#include <stdio.h>
#include <stdlib.h>

int factorial(int n) {
   //base case
   if(n == 0) {
      return 1;
   } else {
      return n * factorial(n-1);
   }
}

int fibbonacci(int n) {
   if(n == 0){
      return 0;
   } else if(n == 1) {
      return 1;
   } else {
      return (fibbonacci(n-1) + fibbonacci(n-2));
   }
}

int main() {
   int n = 30;
   int i;

   int *a = malloc(sizeof(int));
   *a = factorial(10);
	
   printf("a(%x): %d\n", a, *a);
   printf("Factorial of %d: %d\n" , n , factorial(n));
   printf("Fibbonacci of %d: " , n);
	
   for(i = 0;i<n;i++) {
      printf("%d ",fibbonacci(i));            
   }
}
