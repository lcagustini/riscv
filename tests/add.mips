	.text
	.abicalls
	.option	pic0
	.section	.mdebug.abi32,"",@progbits
	.nan	legacy
	.set	mips32r2
	.file	"add.c"
	.text
	.globl	main
	.p2align	2
	.type	main,@function
	.set	nomicromips
	.set	nomips16
	.set	mips32r2
	.ent	main
main:
	.frame	$sp,0,$ra
	.mask 	0x00000000,0
	.fmask	0x00000000,0
	.set	noreorder
	.set	nomacro
	.set	noat
	jr	$ra
	addiu	$2, $zero, 160
	.set	at
	.set	macro
	.set	reorder
	.end	main
$func_end0:
	.size	main, ($func_end0)-main


	.ident	"clang version 4.0.0 (trunk)"
	.section	".note.GNU-stack","",@progbits
	.text
