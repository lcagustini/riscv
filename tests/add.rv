	.file	"add.c"
	.option nopic
	.text
	.section	.text.startup,"ax",@progbits
	.align	2
	.globl	main
	.type	main, @function
main:
	li	a0,88
	ret
	.size	main, .-main
	.ident	"GCC: (GNU) 8.2.0"
