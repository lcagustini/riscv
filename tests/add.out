
add.run:     file format elf32-littleriscv


Disassembly of section .text:

00000100 <_start>:
 100:	00500137          	lui	sp,0x500
 104:	010000ef          	jal	ra,114 <main>
 108:	200002b7          	lui	t0,0x20000
 10c:	000280e7          	jalr	t0 # 20000000 <memory_size>
 110:	00100073          	ebreak

00000114 <main>:
 114:	ff010113          	addi	sp,sp,-16 # 4ffff0 <_RAM_END+0x2ffff0>
 118:	00812623          	sw	s0,12(sp)
 11c:	01010413          	addi	s0,sp,16
 120:	00000793          	li	a5,0
 124:	00078513          	mv	a0,a5
 128:	00c12403          	lw	s0,12(sp)
 12c:	01010113          	addi	sp,sp,16
 130:	00008067          	ret

00000134 <get_id>:
 134:	f1402573          	csrr	a0,mhartid
 138:	00008067          	ret
